app/gui/view/component-browser/searcher-list-panel/src/layouting.rs

```diff
@@ -0,0 +1,159 @@
+//! An algorithm for arranging component groups in the Favorites section in an order that optimizes
+//! the keyboard access to them.
+//!
+//! See [`Layouter`] struct documentation to learn more.
+
+use ensogl_core::prelude::*;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// The layouting algorithm works only for three columns, so this constant is just for
+/// readability, it can't be easily changed.
+const COLUMNS: usize = 3;
+const LEFT: usize = 0;
+const CENTER: usize = 1;
+const RIGHT: usize = 2;
+type Column = usize;
```

Comment:
I think instead of constants and `[T; COLUMNS]` arrays we could have just a parametrized structure with "left", "center", "right" fields.

Reply:
I was considering this idea during the implementation, but the current approach has two advantages:
1. It's easier to iterate over every column in the `column_grid` module when we choose the column for the component group. Iterating over a set of struct fields is ugly.
2. More importantly, the current implementation heavily relies on the fact that we have a few arrays and can address them by column index. For example, `Layouter::push_next_group_to` method takes only a few arguments:
`self.push_next_group_to(LEFT, Some(max_height));`. Changing it to something like `self.left.push_next_group(Some(max_height))` requires passing additional arguments so that it would look more like `self.left.push_next_group(Some(max_height), &mut self.left_height(), &mut self.iter)` which I consider uglier.